### ISSUE TITLE
lua side_name differs from wml side_name

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#4994]
```
    [event]
        name=side 2 turn 1
        [lua]
            code = <<
                local side = wesnoth.sides[wml.variables['side_number']]
                wml.variables['lua_side_name'] = side.side_name
            >>
        [/lua]
        [store_side]
            side=$side_number
            variable=side
        [/store_side]
        [message]
             message="$lua_side_name is not the same as $side.side_name (this one is empty). There exists $side.current_player though."
             speaker=narrator
        [/message]
    [/event]
```

**Comment by gfgtdf** [comment]
https://github.com/wesnoth/wesnoth/commit/29942eec35d266225bc923b5366d849503107053

Lua side_name returns current_player is side_name is empty,  @CelticMinstrel probably knows why

**Comment by CelticMinstrel** [comment]
I can't remember the exact reasoning, but I think it was most likely something like this: the intent was probably for `side_name` to reflect the actual name that is shown in the in-game UI, which means either the name that was explicitly specified in the WML or the default name supplied by the engine.

**Comment by gfgtdf** [comment]
> I can't remember the exact reasoning, but I think it was most likely something like this: the intent was probably for side_name to reflect the actual name that is shown in the in-game UI, which means either the name that was explicitly specified in the WML or the default name supplied by the engine.

Well this does make sense, but on the oter hand it means that codes like `side.side_name = side.side_name` might chaneg side_name which would be unexpected.

**Comment by CelticMinstrel** [comment]
That's true… I suppose the `side_name` setter could have a special case where it sets the name to empty if you're setting it to `current_player`…

**Comment by sevu** [comment]
Whatever there is to do, it should go into 1.16.

**Comment by CelticMinstrel** [comment]
This is now fixed both on 1.16 (839bcb513c44ae34194a30c6e59c27cea69d0537) and on master (70fa5a73d4ea064f1afc47bc535c911291e82fd1).
